### ISSUE TITLE
PAT 2.1.0 datapoint errors using duplicate measure

### ISSUE THREAD

**Issue by shorowit** [NREL/OpenStudio#2565]
See attached project file. 

[resstock_pnw.zip](https://github.com/NREL/OpenStudio/files/914358/resstock_pnw.zip)

It runs correctly as specified. But when I simply click Duplicate Measure twice for the Apply Upgrade measure (resulting in "Apply Upgrade 2" and "Apply Upgrade 2 2") and then re-run the analysis, I get the error below for every datapoint. For some reason, it thinks the duplicated measure should be applied after the EnergyPlus simulation?

Let me know if there's anything else I can provide.

```
[18:37:41.680941 INFO] Attempting to validate the measure workflow
[18:37:41.682169 DEBUG] Validating step 0
[18:37:41.685411 DEBUG] Validated step 0
[18:37:41.685474 DEBUG] Validating step 1
[18:37:41.730405 DEBUG] Validated step 1
[18:37:41.730565 DEBUG] Validating step 2
[18:37:41.732898 DEBUG] Validated step 2
[18:37:41.732948 DEBUG] Validating step 3
[18:37:41.733980 DEBUG] Validated step 3
[18:37:41.734015 DEBUG] Validating step 4
[18:37:41.757311 ERROR] Found error in state 'initialization' with message ["OpenStudio measure /mnt/openstudio/analysis_6afd4e89-e497-4103-8d37-7981fa9d3eb6/measures/ApplyUpgrade called after after Energyplus simulation.:/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/util/measure.rb:121:in `block in validate_measures'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/util/measure.rb:94:in `each'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/util/measure.rb:94:in `each_with_index'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/util/measure.rb:94:in `validate_measures'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/jobs/run_initialization.rb:83:in `perform'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/run.rb:256:in `step'\n/usr/local/bundle/gems/openstudio-workflow-1.2.1/lib/openstudio/workflow/run.rb:210:in `run'\n/opt/openstudio/server/app/workers/run_simulate_data_point.rb:173:in `block in perform'\n/opt/openstudio/server/app/workers/run_simulate_data_point.rb:168:in `open'\n/opt/openstudio/server/app/workers/run_simulate_data_point.rb:168:in `perform'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/performable_method.rb:26:in `perform'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/backend/base.rb:84:in `block in invoke_job'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `block in initialize'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `execute'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:40:in `run_callbacks'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/backend/base.rb:81:in `invoke_job'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:230:in `block (2 levels) in run'\n/usr/local/lib/ruby/2.2.0/timeout.rb:88:in `block in timeout'\n/usr/local/lib/ruby/2.2.0/timeout.rb:98:in `call'\n/usr/local/lib/ruby/2.2.0/timeout.rb:98:in `timeout'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:230:in `block in run'\n/usr/local/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:229:in `run'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:306:in `block in reserve_and_run_one_job'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `block in initialize'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `execute'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:40:in `run_callbacks'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:306:in `reserve_and_run_one_job'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:213:in `block in work_off'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:212:in `times'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:212:in `work_off'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:175:in `block (4 levels) in start'\n/usr/local/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:174:in `block (3 levels) in start'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `block in initialize'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `execute'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:40:in `run_callbacks'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:173:in `block (2 levels) in start'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:172:in `loop'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:172:in `block in start'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/plugins/clear_locks.rb:7:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/plugins/clear_locks.rb:7:in `block (2 levels) in '\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:79:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:79:in `block (2 levels) in add'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:61:in `block in initialize'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:79:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:79:in `block in add'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `call'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:66:in `execute'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/lifecycle.rb:40:in `run_callbacks'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/worker.rb:171:in `start'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/command.rb:132:in `run'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/command.rb:120:in `block in run_process'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/application.rb:266:in `call'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/application.rb:266:in `block in start_proc'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/application.rb:275:in `call'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/application.rb:275:in `start_proc'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/application.rb:296:in `start'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/controller.rb:59:in `run'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons.rb:197:in `block in run_proc'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/cmdline.rb:92:in `call'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons/cmdline.rb:92:in `catch_exceptions'\n/usr/local/bundle/gems/daemons-1.2.4/lib/daemons.rb:196:in `run_proc'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/command.rb:118:in `run_process'\n/usr/local/bundle/gems/delayed_job-4.1.2/lib/delayed/command.rb:94:in `daemonize'\nbin/delayed_job:6:in `
'"]} 
```

**Comment by asparke2** [comment]
This is a blocker for ResStock to use OS 2.0

**Comment by DavidGoldwasser** [comment]
May be duplicate of issue #2549

**Comment by rHorsey** [comment]
This issue appears to be due to the workflow reorganization method not being invoked during the measure step duplication process. As a result, measure steps are appended to the end of the workflow, as expected, but not reorganized into the correct order, as would be expected. This causes the workflow validation code to error out, as Ruby measures are called after the EnergyPlus run should have taken place.

**Comment by asparke2** [comment]
@henryHorsey Is the fix something that needs to happen in PAT, or on the server?

**Comment by DavidGoldwasser** [comment]
@asparke2 I think this is at least in part in PAT because I see issue before running the job. See screenshots in #2549

**Comment by asparke2** [comment]
This was just hit by an EDAPT user working on a project.  They couldn't figure out what the issue was.

**Comment by asparke2** [comment]
Confirmed on development, should be fixed in 2.1.2 installer
